tests/early-data.rs

```diff
@@ -75,38 +95,33 @@ async fn test_0rtt_impl(vectored: bool) -> io::Result<()> {
         .unwrap();
     server.max_early_data_size = 8192;
     let server = Arc::new(server);
+    let acceptor = Arc::new(TlsAcceptor::from(server));
 
-    let listener = TcpListener::bind("127.0.0.1:0")?;
+    let listener = TcpListener::bind("127.0.0.1:0").await?;
     let server_port = listener.local_addr().unwrap().port();
-    thread::spawn(move || loop {
-        let (mut sock, _addr) = listener.accept().unwrap();
+    tokio::spawn(async move {
+        loop {
+            let (mut sock, _addr) = listener.accept().await.unwrap();
+
+            let acceptor = acceptor.clone();
+            tokio::spawn(async move {
+                let stream = acceptor.accept(&mut sock).await.unwrap();
 
-        let server = Arc::clone(&server);
-        thread::spawn(move || {
-            let mut conn = ServerConnection::new(server).unwrap();
-            conn.complete_io(&mut sock).unwrap();
+                let mut buf = Vec::new();
+                let mut stream_wrapper = TlsStreamEarlyWrapper { inner: stream };
+                stream_wrapper.read_to_end(&mut buf).await.unwrap();
+                let mut stream = stream_wrapper.inner;
+                stream.write_all(b"EARLY:").await.unwrap();
```

Comment:
Should there be a test that explicitly tests for being able to read actual data off of the early data stream?

Reply:
I don't follow. This is already reading the early data?